File: Lodjinha/app/src/main/java/br/com/bsavoini/lodjinha/home/CategoriesAdapter.java
package br.com.bsavoini.lodjinha.home;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import br.com.bsavoini.lodjinha.R;
import br.com.bsavoini.lodjinha.api.model.CategoryModel;
import com.squareup.picasso.Picasso;


import java.util.List;

public class CategoriesAdapter extends RecyclerView.Adapter<CategoriesAdapter.CategoryViewHolder> {
    private List<CategoryModel> categoriesArr;
    private Picasso picassoInstance;

    public CategoriesAdapter(List<CategoryModel> categoriesArr) {
        this.categoriesArr = categoriesArr;
    }

    @NonNull
    @Override
    public CategoryViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        picassoInstance = Picasso.with(viewGroup.getContext());
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_category, viewGroup, false);
        return new CategoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CategoryViewHolder viewHolder, int i) {
        CategoryModel categoryModel = categoriesArr.get(viewHolder.getAdapterPosition());
        viewHolder.categoryTxt.setText(categoryModel.getDescription());

        picassoInstance.load(categoryModel.getImageURL())
                .placeholder(R.drawable.logo_navbar)
                .into(viewHolder.categoryImg);

    }

    @Override
    public int getItemCount() {
        return categoriesArr.size();
    }

    class CategoryViewHolder extends RecyclerView.ViewHolder {
        private TextView categoryTxt;
        private ImageView categoryImg;

        CategoryViewHolder(@NonNull View itemView) {
            super(itemView);
            categoryTxt = itemView.findViewById(R.id.txt_category);
            categoryImg = itemView.findViewById(R.id.img_category);
        }
    }
}

File: app/src/main/java/com/example/lidjinha/lodjinha/home/CategoriesAdapter.kt
package com.example.lidjinha.lodjinha.home

import android.content.Context
import android.support.v4.content.ContextCompat
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.example.lidjinha.lodjinha.R
import com.example.lidjinha.lodjinha.model.Categorie
import com.squareup.picasso.Callback
import com.squareup.picasso.Picasso

class CategoriesAdapter(private val categories: List<Categorie>,
                        private val context: Context) :
        RecyclerView.Adapter<CategoriesAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_categories_list, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        Picasso.with(context).load(categories[position].urlImage).into(holder.image, callbackImage(holder.image))
        holder.description.text = categories[position].description
    }

    private fun callbackImage(image: ImageView) = object : Callback {
        override fun onSuccess() {
            Log.d("Load categories", "Success!")
        }

        override fun onError() {
            image.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.ic_down_black))
        }
    }

    override fun getItemCount(): Int {
        return categories.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val image: ImageView = itemView.findViewById(R.id.iv_categorie_image)
        val description: TextView = itemView.findViewById(R.id.tv_categorie_description)

    }
}